web/js/sidebar/ui.js

```diff
@@ -35,7 +35,7 @@ export function sidebarUi(models, config, ui) {
       trailing: true
     });
     var layerAdd = function(layer) {
-      if (!ui.tour.resetting && !ui.naturalEvents.selecting) {
```

Comment:
@ZachTRice It looks like `sidebar` component is rendering many times on each step change, slowing things down. I would suggest something like this being applied on step change, just allowing the component to render one time after the state of everything else has been updated.
<img width="1049" alt="screen shot 2019-01-31 at 11 24 55 am" src="https://user-images.githubusercontent.com/3605988/52068943-bb7fca80-254b-11e9-8a97-1ce5a8249ba4.png">
`Redux` implementation should prevent us from needing to this kind of hacky event-listener workaround 🤞

Reply:
I think this is going to need to re-render each time because it is loading a new link (state) with a group of layers. So, each step, the layers/model/load method is called with that currentState passed in which includes the group of layers. When there are, for example, 8 layers in a link, self.add is ran for each and updateLayers() is then called 8 times. I'm not sure how I would get around this, as it needs to make sure each layer is in the sidebar with the correct properties enabled, hidden, opacity, colormaps etc.